server/src/main/java/org/opensearch/common/settings/DoubleParser.java

```diff
@@ -0,0 +1,99 @@
+package org.opensearch.common.settings;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.function.Function;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.settings.Setting.Property;
+import org.opensearch.common.settings.Setting.Validator;
+
+public class DoubleParser<T> implements Function<String, T>, Writeable {
```

Comment:
If you make `DoubleParser` an inner class inside `Setting`, then:
 - You won't need to include the generic type `<T>` here
 - You will automatically gain access to the `parseDouble()` method inside the `Setting` class so you don't have to recreate it yourself.  That will really simplify the rest of your implementation!

Reply:
Agree, I will make `DoubleParser` as an inner class inside `Setting`